.github/workflows/build-test.yml

```diff
@@ -80,9 +80,40 @@ jobs:
       - name: Fetch tags
         run: |
           git fetch --tags
+          
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+        
+      - name: Cache Composer dependencies
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+            
+      - name: Install PHP_CodeSniffer
+        run: |
+          curl -sS https://getcomposer.org/installer | php
+          php composer.phar require --dev squizlabs/php_codesniffer
+          
+      - name: Run PHP_CodeSniffer
+        run: |
+          ./vendor/bin/phpcs --standard=PSR12 src/ --ignore=*/tests/*
 
+      - name: Install PHPStan
+        run: |
+          curl -sS https://getcomposer.org/installer | php
+          php composer.phar require --dev phpstan/phpstan
+          
+      - name: Run PHPStan
+        run: |
+          ./vendor/bin/phpstan analyse src --level=max
+    
```

Comment:
We have a separate stage for Static Checks and Code Sniffing. You can look into it here: [static-checks.yml](https://github.com/fossology/fossology/blob/master/.github/workflows/static-checks.yml)